zipline/pipeline/loaders/events.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import pandas as pd
+from six import iteritems
+from toolz import merge
+
+from .base import PipelineLoader
+from .frame import DataFrameLoader
+from .utils import next_date_frame, previous_date_frame, previous_value
+
+TS_FIELD_NAME = "timestamp"
+SID_FIELD_NAME = "sid"
+
+
+class EventsLoader(PipelineLoader):
+    """
+    Abstract loader.
+
+    Does not currently support adjustments to the dates of known events.
+
+    Parameters
+    ----------
+    all_dates : pd.DatetimeIndex
+        Index of dates for which we can serve queries.
+    events_by_sid : dict[int -> pd.DataFrame]
+        Dict mapping sids to DataFrames representing dates on which events
+        occurred along with other associated values.
+
+        If the DataFrames contain a "timestamp" column, that column is
+        interpreted as the date on which we learned about the event.
+
+        If the DataFrames do not contain a "timestamp" column, we assume we
+        knew about the event on all prior dates.  This mode is only supported
+        if ``infer_timestamp`` is explicitly passed as a truthy value.
+    infer_timestamps : bool, optional
+        Whether to allow omitting the "timestamp" column.
+    dataset : DataSet
+        The DataSet object for which this loader loads data.
+    expected_cols : frozenset
+        Set of expected columns for the dataset, without timestamp.
+    """
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=None,
+                 expected_cols=frozenset()):
```

Comment:
I think this would make more sense as a class attribute. Users should not make two instances of the same class that expect different columns